### PR TITLE
rustup https://github.com/rust-lang/rust/pull/66877

### PR DESCRIPTION
**Pull request by matthiaskrgr**
changelog: none

### CODE REVIEW

File: clippy_lints/src/consts.rs [hunk 1]
```diff
@@ -345,7 +345,7 @@ impl<'c, 'cc> ConstEvalLateContext<'c, 'cc> {
                     promoted: None,
                 };
 
-                let result = self.lcx.tcx.const_eval(self.param_env.and(gid)).ok()?;
+                let result = self.lcx.tcx.const_eval_validated(self.param_env.and(gid)).ok()?;
```
Comment (oli-obk) [inline clippy_lints/src/consts.rs]: Use `self.lcx.tcx.const_eval_resolve(self.param_env, def_id, substs, None)`

**Comment by JohnTitor** [comment]
I think we should use `const_eval_resolve` and `const_eval_poly` instead so that we can reduce code and avoid using `const_eval_validated` (documentation says "**Do not use this** directly"), opened #4944 for an alternative.

File: clippy_lints/src/enum_clike.rs [hunk 2]
```diff
@@ -56,7 +56,7 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for UnportableVariant {
                         instance,
                         promoted: None,
                     };
-                    let constant = cx.tcx.const_eval(param_env.and(c_id)).ok();
+                    let constant = cx.tcx.const_eval_validated(param_env.and(c_id)).ok();
```
Comment (oli-obk) [inline clippy_lints/src/enum_clike.rs]: use `cx.tcx.const_eval_poly(def_id)`

**Comment by oli-obk** [comment]
closing in favour of https://github.com/rust-lang/rust-clippy/pull/4944

**Pull request closed without merging**